README.md

```diff
@@ -60,3 +95,44 @@ Then use the `analyticsClient` variable:
+We use this prefixing method to identify and group events in the final destination for organization purposes and helping the analysts traverse the events with ease.
+This is especially useful when there are multiple components sending events to the same project in the final destination. To give an example, our marketing website, docs pages and blog all send `Page View` event but these are prefixed respectively `Marketing`, `Docs` and `Blog`. So the final event name is `[Marketing] Page View`, `[Docs] Page View`, etc. Furthermore, this method makes it easier to identify all the `Marketing` events.
+
+## Identifying users
+
+By default the client tracks `deviceId` to identify the current device of the end user. `deviceId` is a UUID generated automatically by the client. Use `setUserId` method to identify authenticated user. At least one of these identifiers is required to successfully track an event.
+
+In most cases the device refers to the browser of the user (or Electron application for the case of Etcher). By persisting the `deviceId` in the local storage, we are able to identify the events coming from the same user in order to create a timeline of component usage.
+
+It is possible to generate a new device identifier and also override an existing device identifier. Additionally, when we record multiple device identifiers along with the same user identifier, we are able to link multiple event timelines coming from various devices belonging to the same user.
+For instance, a balenaCloud user might login to the dashboard from multiple browsers and also use balenaCLI. We would like to create a coherent timeline that captures all their activity and display it under the same user in order to make a sound analysis of their balena platform usage. As long as all the components `setUserId` as soon as the authentication happens, we are able to link all the events with separate `deviceId` values that came *even before* the authentication on the backend side.
+
+[`AnalyticsUrlParams`](src/url-params.ts) helps with linking anonymous `deviceId` values that each component sends individually. For instance we use this method to link Etcher users clicking on EFP link to land on our blog website and eventually signup to balenaCloud. When the user clicks on the blog link in EFP, we append the `deviceId` generated by EFP component to the URL. Then the blog parses the URL to match EFP `deviceId` with blog `deviceId`. In a similar fashion, the signup links on the blog website also includes EFP `deviceId` as well as blog `deviceId`. When the user lands on balenaCloud dashboard and authenticates, we link all the user activity and list it under the same timeline. We need to make these jumps as there are 3 components tracking events and there are multiple domains involved. The benefit is we are able to look into a user's EFP and blog activity before signing up to balenaCloud.
```

Comment:
As I understand, it means pretty tight integration with specific components appending extra information to the URL, and then only specific components at particular endpoints are expecting and parsing that extra information.
Is that an accurate description of our use case?

Reply:
BTW, maybe "blog `deviceId`" could be easier understood as "browser `deviceId`", no?

Reply:
Indeed it's a pretty tight integration Pranas. As we discussed last week with Stevche as well, it makes sense to improve the current state of implementation to make the whole device identifier linking flow more convenient for component maintainers.